### PR TITLE
irinterp: clean up IR abstract interpretation subroutines [NFC]

### PR DESCRIPTION
**Pull request by aviatesk**

### CODE REVIEW

File: base/compiler/ssair/irinterp.jl [hunk 1]
```diff
@@ -182,60 +177,49 @@ function reprocess_instruction!(interp::AbstractInterpreter, ir::IRCode, mi::Met
...
+    if rt !== nothing && !⊑(typeinf_lattice(interp), typ, rt)
+        ir.stmts[idx][:type] = rt
+        return true
     end
     return false
 end
 
 function _ir_abstract_constant_propagation(interp::AbstractInterpreter, mi_cache,
-        mi::MethodInstance, ir::IRCode, argtypes::Vector{Any}; extra_reprocess = nothing)
+        mi::MethodInstance, ir::IRCode, argtypes::Vector{Any})
```
Comment (Keno) [inline base/compiler/ssair/irinterp.jl:222]: `extra_reprocess` is used in packages.

Comment (aviatesk) [reply base/compiler/ssair/irinterp.jl]: Okay, I recovered it back.

File: base/compiler/ssair/irinterp.jl [hunk 2]
```diff
@@ -108,11 +84,55 @@ function getindex(tpdum::TwoPhaseDefUseMap, idx::Int)
...
+        argtypes = va_process_argtypes(argtypes, mi)
+        argtypes_refined = Bool[!⊑(typeinf_lattice(interp), ir.argtypes[i], argtypes[i]) for i = 1:length(argtypes)]
+        empty!(ir.argtypes)
+        append!(ir.argtypes, argtypes)
+        tpdum = TwoPhaseDefUseMap(length(ir.stmts))
+        ssa_refined = BitSet()
+        lazydomtree = LazyDomtree(ir)
+        return new(ir, mi, world, argtypes_refined, tpdum, ssa_refined, lazydomtree)
+    end
+end
```
Comment (aviatesk) [inline base/compiler/ssair/irinterp.jl:106]: I setup this `IRInterpretationState` that does a similar role as `InferenceState` in order to simplify the signatures of irinterp subroutines. @Keno does it seem fine to you?

**Comment by aviatesk** [comment]
@nanosoldier `runbenchmarks("inference", vs=":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/46917#issuecomment-1260564328) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/ca1ea2d_vs_1afa368/report.md).

**Comment by aviatesk** [comment]
@nanosoldier `runbenchmarks("inference", vs=":master")`

File: base/compiler/ssair/irinterp.jl (hunk 2 above)
Comment (Keno) [reply base/compiler/ssair/irinterp.jl:106]: Sure, seems fine.

**Review by Keno — APPROVED** [review]

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/46917#issuecomment-1261644195) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/d057b15_vs_75e96c5/report.md).

**Comment by aviatesk** [comment]
Regressions seem to be minor. The improvements in terms of code simplicity wins in my opinion.

**Pull request merged**